packages/repository/src/__tests__/integration/repositories/resolve-belongs-to-metadata.integration.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  BelongsToDefinition,
+  Entity,
+  ModelDefinition,
+  RelationType,
+} from '../../..';
+import {resolveBelongsToMetadata} from '../../../relations/belongs-to/belongs-to.helpers';
+
+describe('resolveBelongsToMetadata', () => {
+  it('throws if the wrong metadata type is used', async () => {
+    let error;
+
+    try {
+      resolveBelongsToMetadata(Category.definition.relations[
+        'category'
+      ] as BelongsToDefinition);
+    } catch (err) {
+      error = err;
+    }
+
+    expect(error.message).to.eql(
```

Comment:
The `error` may be `undefined`. Please use `expect(() => { code to throw }).to.throw( error message or regexp )` instead.

Reply:
```ts
expect( () => {
    resolveBelongsToMetadata(Category.definition.relations[
        'category'
      ] as BelongsToDefinition);
    }).to.throw('Invalid hasOne definition for Category#category: relation type must be BelongsTo');
```